webextension/css/popup.css

```diff
@@ -636,6 +647,32 @@ span ~ .panel-header-text {
   padding-inline-start: 16px;
 }
 
+#edit-sites-assigned {
+  flex: 1;
+}
+
+#edit-sites-assigned h3 {
+  font-size: 14px;
+  font-weight: normal;
+  padding-block-end: 5px;
+  padding-inline-end: 16px;
+  padding-inline-start: 16px;
+}
+
+#edit-sites-assigned table td {
+  display: flex;
+  padding-inline-end: 16px;
+  padding-inline-start: 16px;
+}
+
+#edit-sites-assigned .delete-assignment {
+  display: none;
+}
+
+#edit-sites-assigned td:hover > .delete-assignment {
```

Comment:
This `>` child-combinator selector means only hovering on the host name will show the delete icon, which I didn't discover at first. Maybe add a class to the `tr` element like `site-assignment` and use:
```
tr.site-assignment .delete-assignment {
  display: block;
}
tr.site-assignment:hover .delete-assignment {
  display: block;
}
```
?